src/uu/date/src/date.rs

```diff
@@ -227,8 +227,20 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
             DateSource::Human(relative_time) => {
                 // Get the current DateTime<FixedOffset> for things like "1 year ago"
                 let current_time = DateTime::<FixedOffset>::from(Local::now());
-                let iter = std::iter::once(Ok(current_time + relative_time));
-                Box::new(iter)
+                // double check the result is overflow or not of the current_time + relative_time
+                // it may cause a panic of chrono::datetime::DateTime add
+                match current_time.checked_add_signed(relative_time) {
+                    Some(date) => {
+                        let iter = std::iter::once(Ok(date));
+                        Box::new(iter)
+                    }
+                    None => {
+                        return Err(USimpleError::new(
+                            1,
+                            format!("invalid date {:?}", relative_time),
```

Comment:
I'm just now noticing this, using Debug for user-facing errors is not optimal. I think there's another `invalid date` message you can just copy.